src/main/java/storage/TaskList.java

```diff
@@ -0,0 +1,158 @@
+package storage;
+
+import java.util.ArrayList;
+
+import commands.Command;
+import exceptions.InvalidTaskException;
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.Todo;
+
+/**
+ * Stores the list of tasks.
+ * Responsible for manipulating tasks, such as adding, marking, unmarking and deleting.
+ * Responsible for printing out all the tasks in the list.
+ */
+public class TaskList {
+    private final ArrayList<Task> tasks;
+
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Adds a task to TASKS.
+     *
+     * @param type type of task to be added.
+     * @param response arguments that the task has.
+     * @return the task that has been added to TASKS.
+     */
+    public Task addTask(Command type, String[] response) {
+        Task task = new Task(response[0]);
+        switch (type) {
+        case TODO:
+            task = new Todo(response[1]);
+            break;
+        case DEADLINE:
+            // response[0] = description, response[1] = deadline
+            task = new Deadline(response[0], response[1].strip());
+            break;
+        case EVENT:
+            String[] times = response[1].split("/to ");
+            task = new Event(response[0], times[0].strip(), times[1].strip());
+            break;
+        default:
+            break;
+        }
+        tasks.add(task);
+        return task;
+    }
+
```

Comment:
The addTask method in the TaskList class handles both task creation and addition. While this approach works, it might be worth considering separating the task creation logic into a dedicated method or class. This would enhance the Single Responsibility Principle (SRP), making the code easier to maintain, especially if task creation logic becomes more complex in the future. This separation could also make the method cleaner and more focused on adding tasks to the list. What do you think about this?